packages/bsync/src/subscriptions/revenueCatWebhookHandler.ts

```diff
@@ -0,0 +1,174 @@
+import { IncomingMessage, ServerResponse } from 'node:http'
+import AppContext from '../context'
+import { isValidDid } from '../routes/util'
+import { Database } from '..'
+import { sql } from 'kysely'
+import { createSubsOpChannel } from '../db/schema/subs_op'
+
+export const isRevenueCatWebhookUrl = (urlStr: string | undefined) => {
+  if (!urlStr) return false
+  const url = new URL(urlStr, 'http://host')
+  return url.pathname === '/webhooks/revenuecat'
+}
+
+const parseBody = async (req: IncomingMessage) =>
+  new Promise((resolve, reject) => {
+    let body = ''
+    req.on('data', (chunk) => (body += chunk))
+    req.on('end', () => {
+      try {
+        resolve(JSON.parse(body))
+      } catch (err) {
+        reject(err)
+      }
+    })
+    req.on('error', reject)
+  })
```

Comment:
Two things here:
a. if the response is compressed then this would be the compressed body.
b. could probably simplify this and avoid worrying about e.g. setting-up and cleaning-up event handlers:

```js
const chunks: Buffer[] = []
for await (const chunk of req) {
  chunks.push(chunk)
}
return Buffer.concat(chunks).toString()
```

Reply:
I think `compression` may just be for compression of responses, but I don't see the harm in it so long as it doesn't conflict with the connect middleware.